fdbclient/include/fdbclient/BlobCipher.h

```diff
@@ -95,7 +98,12 @@ class BlobCipherMetrics : public NonCopyable {
 
 	static CounterSet& counters(UsageType t) {
 		ASSERT(t < UsageType::MAX);
-		return getInstance()->counterSets[int(t)];
+		auto& counterSets = getInstance()->counterSets;
+		int index = FLOW_KNOBS->ENCRYPT_KEY_CACHE_ENABLE_DETAIL_LOGGING ? (int)t : (int)UsageType::ALL;
```

Comment:
nit: does it mean, if knob is NOT selected, then, all counters logging is enabled? If yes, shouldn't it be other way round?

Reply:
If the knob is true then all counters would be logged. It is false by default

Reply:
Got it. I think we have `ENABLE_ENCRYPTION_CPU_TIME_LOGGING` knob to control if details needs to be logged. 

Alternate desirable feature could be: selectively control what all components could log the counters, for instance: if we define it as a bitmap:

```
0          -> NONE
1 << 1   -> TLog
1 << 2   -> CP
1 << 3   -> Blob
1 << 4   -> Backup
```

For instance: if I want individual components I can select above mentioned number, however, any combination will be `||` of component combinations.

Reply:
the `ENABLE_ENCRYPTION_CPU_TIME_LOGGING` controlled whether we log the cipher cpu time, which worried it may come with overhead, though we never confirm that's the case. but even when it is disabled, we will be logging KMS request latencies for each usage types. The new `ENCRYPT_KEY_CACHE_ENABLE_DETAIL_LOGGING` will combine the cpu time logging and the KMS request latencies metrics across usage types into an "overall" usage type, to reduce number of counters. The two knobs are orthogonal.

fine control of which counters to log would be an overkill to me.

Reply:
> The two knobs are orthogonal.

Got it. Thanks for explanation. I would still encourage we think about minimizing KNOBs unless they are absolutely necessary. For instance: the improvement done in this PR are `absolute must have` may be we update default behavior than adding new KNOB? OR atleast have a TODO/FIXME in the code to do that change sometime soon. 

> fine control of which counters to log would be an overkill to me.

I think its pretty useful, it is quite possible in production environment to focus on some specific codepath and enable the logging counters to triage it further; for instance: we want to perf benchmark EaR and enabling TLog counter while running 100% write might be sufficient, same is applicable while triaging high latencies observation for commits OR reads for some reason.

Reply:
suggestions make sense, but how about we do them in a follow up PR?

Reply:
Sounds great. Thanks

Reply:
It's also reasonable that if we are concerned about EKP latency in specific situations we can also instrument them as part of that situation directly instead of building a more complex framework here too

Reply:
As we discussed in a separate avenue though, pretty much all of these counters only happen on one process
The only process that can have multiple types iirc is the TLog with both the TLog and TLogPostResolution counters